java/vector/src/main/java/org/apache/arrow/vector/VarCharVector.java

```diff
@@ -270,6 +273,21 @@ public void setSafe(int index, Text text) {
     setSafe(index, text.getBytes(), 0, text.getLength());
   }
 
+  @Override
+  public int hashCode (int index) {
+    //TODO cal hashCode in memory level
+    byte[] values = get(index);
+    return Arrays.hashCode(values);
+  }
+
+  @Override
+  public boolean equals(int index, ValueVector to, int toIndex) {
+    //TODO compare value in memory level
+    String value1 = this.getObject(index).toString();
+    String value2 = to.getObject(toIndex).toString();
```

Comment:
Also, there are unnecessary memory copies here.

Reply:
Sure, this should be done use something like ByteFunctionHelpers#equals